### PR TITLE
feat(stdlib): Add number constants to number libraries

### PR DESCRIPTION
**Pull request by spotandjake**
For #1017, adds `Number.pi`, `Number.tau`, `Number.e`, `Float32.pi`, `Float32.tau`, `Float32.e`, `Float64.pi`, `Float64.tau`, `Float64.e`

### CODE REVIEW

File: compiler/test/stdlib/float32.test.gr [hunk 1]
```diff
@@ -21,3 +21,10 @@ assert gt(infinity, 100000000.f)
 assert toNumber(infinity) == toNumber(infinity)
 assert toNumber(infinity) == (toNumber(infinity) - 1)
 assert nan != nan
+
+// Constants Test
+assert pi == 3.1415927f
+
+assert tau == 3.1415927f
+
+assert e == 2.7182817f
```
Comment (peblair) [inline compiler/test/stdlib/float32.test.gr:30]: @spotandjake can you add the newlines at the end of these test files?

**Review by peblair — CHANGES_REQUESTED** [review]
🔥 Thanks, Jake! One minor nitpick, but this is very helpful.

File: compiler/test/stdlib/float32.test.gr [hunk 2]
```diff
@@ -1,5 +1,19 @@
...
+// smoke test:
+assert gt(infinity, 100000000.f)
+// test infinity-specific semantics:
+assert toNumber(infinity) == toNumber(infinity)
+assert toNumber(infinity) == (toNumber(infinity) - 1)
+assert nan != nan
+
+assert pi == 3.1415927f
+
+assert tau == 3.1415927f
```
Comment (ospencer) [inline compiler/test/stdlib/float32.test.gr:13]:
```suggestion
assert tau == 6.2831853f
```

File: compiler/test/stdlib/float64.test.gr [hunk 3]
```diff
@@ -1,5 +1,18 @@
...
+// Constants Tests
+// smoke test:
+assert gt(infinity, 100000000.d)
+// test infinity-specific semantics:
+assert toNumber(infinity) == (toNumber(infinity) - 1)
+assert nan != nan
+
+assert pi == 3.141592653589793d
+
+assert tau == 3.141592653589793d
```
Comment (ospencer) [inline compiler/test/stdlib/float64.test.gr:12]:
```suggestion
assert tau == 6.283185307179586d
```

File: compiler/test/stdlib/number.test.gr [hunk 4]
```diff
@@ -6,6 +6,13 @@ import Float32 from "float32"
 import Float64 from "float64"
 import BI from "bigint"
 
+// Constants Test
+assert Number.pi == 3.1415927
+
+assert Number.tau == 3.1415927
```
Comment (ospencer) [inline compiler/test/stdlib/number.test.gr:12]:
```suggestion
assert Number.tau == 6.283185307179586
```

File: compiler/test/stdlib/number.test.gr [hunk 5]
```diff
@@ -6,6 +6,13 @@ import Float32 from "float32"
 import Float64 from "float64"
 import BI from "bigint"
 
+// Constants Test
+assert Number.pi == 3.1415927
```
Comment (ospencer) [inline compiler/test/stdlib/number.test.gr:10]:
```suggestion
assert Number.pi == 3.141592653589793
```

File: stdlib/float32.gr [hunk 6]
```diff
@@ -13,6 +13,56 @@ import {
...
+ * @since v0.5.2
+ */
+export let pi = 3.1415927f
+
+/**
+ * TAU represented as a Float32 value.
+ * 
+ * @since v0.5.2
+ */
+export let tau = 3.1415927f
```
Comment (ospencer) [inline stdlib/float32.gr:58]:
```suggestion
export let tau = 6.2831853f
```

File: stdlib/float32.gr [hunk 7]
```diff
@@ -13,6 +13,56 @@ import {
...
+@unsafe
+export let nan = {
+  let ptr = newFloat32(
+    WasmF32.reinterpretI32(0b01111111100000000000000000000001n)
+  )
+  WasmI32.toGrain(ptr): Float32
+}
+
+/**
+ * PI represented as a Float32 value.
```
Comment (ospencer) [inline stdlib/float32.gr:47]:
```suggestion
 * Pi represented as a Float32 value.
```

File: stdlib/float32.gr [hunk 8]
```diff
@@ -13,6 +13,56 @@ import {
...
+ * 
+ * @since v0.5.2
+ */
+export let pi = 3.1415927f
+
+/**
+ * TAU represented as a Float32 value.
```
Comment (ospencer) [inline stdlib/float32.gr:54]:
```suggestion
 * Tau represented as a Float32 value.
```

File: stdlib/float32.gr [hunk 9]
```diff
@@ -13,6 +13,56 @@ import {
...
+ * 
+ * @since v0.5.2
+ */
+export let tau = 3.1415927f
+
+/**
+ * Eulers Number represented as a Float32 value.
```
Comment (ospencer) [inline stdlib/float32.gr:61]:
```suggestion
 * Euler's number represented as a Float32 value.
```

File: stdlib/float64.gr [hunk 10]
```diff
@@ -13,6 +13,61 @@ import {
...
+  let ptr = newFloat64(
+    WasmF64.reinterpretI64(
+      0b0111111111110000000000000000000000000000000000000000000000000001N
+    )
+  )
+  WasmI32.toGrain(ptr): Float64
+}
+
+/**
+ * PI represented as a Float64 value.
```
Comment (ospencer) [inline stdlib/float64.gr:51]:
```suggestion
 * Pi represented as a Float64 value.
```

File: stdlib/float64.gr [hunk 11]
```diff
@@ -13,6 +13,61 @@ import {
...
+ * 
+ * @since v0.5.2
+ */
+export let pi = 3.141592653589793d
+
+/**
+ * TAU represented as a Float64 value.
```
Comment (ospencer) [inline stdlib/float64.gr:58]:
```suggestion
 * Tau represented as a Float64 value.
```

File: stdlib/float64.gr [hunk 12]
```diff
@@ -13,6 +13,61 @@ import {
...
+ * 
+ * @since v0.5.2
+ */
+export let tau = 3.141592653589793d
```
Comment (ospencer) [inline stdlib/float64.gr:62]:
```suggestion
export let tau = 6.283185307179586d
```

File: stdlib/float64.gr [hunk 13]
```diff
@@ -13,6 +13,61 @@ import {
...
+
+/**
+ * Eulers Number represented as a Float64 value.
```
Comment (ospencer) [inline stdlib/float64.gr:65]:
```suggestion
 * Euler's number represented as a Float64 value.
```

File: stdlib/number.gr [hunk 14]
```diff
@@ -19,6 +19,34 @@ import { parseInt } from "runtime/stringUtils"
 import { newFloat64, newInt64 } from "runtime/dataStructures"
 import Tags from "runtime/unsafe/tags"
 
+/**
+ * @section Constants: Number constant values.
+ */
+/**
+ * PI represented as a Number value.
```
Comment (ospencer) [inline stdlib/number.gr:26]:
```suggestion
 * Pi represented as a Number value.
```

File: stdlib/number.gr [hunk 15]
```diff
@@ -19,6 +19,34 @@ import { parseInt } from "runtime/stringUtils"
...
+ * @since v0.5.2
+ */
+export let pi = 3.1415927
+
+/**
+ * TAU represented as a Number value.
+ * 
+ * @since v0.5.2
+ */
+export let tau = 3.1415927
```
Comment (ospencer) [inline stdlib/number.gr:37]:
```suggestion
export let tau = 6.283185307179586
```

File: stdlib/number.gr [hunk 16]
```diff
@@ -19,6 +19,34 @@ import { parseInt } from "runtime/stringUtils"
...
+ */
+/**
+ * PI represented as a Number value.
+ * 
+ * @since v0.5.2
+ */
+export let pi = 3.1415927
+
+/**
+ * TAU represented as a Number value.
```
Comment (ospencer) [inline stdlib/number.gr:33]:
```suggestion
 * Tau represented as a Number value.
```

File: stdlib/number.gr [hunk 17]
```diff
@@ -19,6 +19,34 @@ import { parseInt } from "runtime/stringUtils"
...
 
+/**
+ * @section Constants: Number constant values.
+ */
+/**
+ * PI represented as a Number value.
+ * 
+ * @since v0.5.2
+ */
+export let pi = 3.1415927
```
Comment (ospencer) [inline stdlib/number.gr:30]:
```suggestion
export let pi = 3.141592653589793
```

File: stdlib/number.gr [hunk 18]
```diff
@@ -19,6 +19,34 @@ import { parseInt } from "runtime/stringUtils"
...
+
+/**
+ * TAU represented as a Number value.
+ * 
+ * @since v0.5.2
+ */
+export let tau = 3.1415927
+
+/**
+ * Eulers Number represented as a Number value.
```
Comment (ospencer) [inline stdlib/number.gr:40]:
```suggestion
 * Euler's number represented as a Number value.
```

File: compiler/test/stdlib/number.test.gr [hunk 19]
```diff
@@ -6,6 +6,13 @@ import Float32 from "float32"
...
+
+assert Number.tau == 3.1415927
+
+assert Number.e == 2.7182817
```
Comment (ospencer) [inline compiler/test/stdlib/number.test.gr:14]:
```suggestion
assert Number.e == 2.718281828459045
```

File: stdlib/number.gr [hunk 20]
```diff
@@ -19,6 +19,34 @@ import { parseInt } from "runtime/stringUtils"
...
+ * 
+ * @since v0.5.2
+ */
+export let e = 2.7182817
```
Comment (ospencer) [inline stdlib/number.gr:44]:
```suggestion
export let e = 2.718281828459045
```

**Review by ospencer — CHANGES_REQUESTED** [review]
I noticed that the value for tau might have been copied and pasted from pi, so I updated that here. Also, since `Number` uses f64s under the hood, I updated pi, tau, and e to be the higher precision variants 😄

File: compiler/test/stdlib/float32.test.gr (hunk 2 above)
Comment (spotandjake) [reply compiler/test/stdlib/float32.test.gr:13]: oops

**Review by ospencer — APPROVED** [review]
Great work @spotandjake!

**Pull request merged**